pagefaultgames/pokerogue
# [Move] Add Retaliate double damage condition

## What are the changes?

The retaliate move can now deal double damage if an ally died the previous turn.
If you kill an enemy and another enemy use retaliate the next turn, the damages will be doubled.

## Why am I doing these changes?

Make retaliate double damage condition works.



## What did change?


I added those 2 attributes in the class Battle :
```typescript
public turnsSincePlayerFaints: number; // The amount of turns after pokemon on the players side have fainted
public turnsSinceEnemyFaints: number; // The amount of turns after pokemon on the players side have fainted
```

The move : 

```typescript
new AttackMove(Moves.RETALIATE, Type.NORMAL, MoveCategory.PHYSICAL, 70, 100, 5, -1, 0, 5)
      .attr(MovePowerMultiplierAttr, (user, target, move) => (user.scene.currentBattle.turnsSincePlayerFaints === 1 && user instanceof PlayerPokemon || user.scene.currentBattle.turnsSinceEnemyFaints === 1 && user instanceof EnemyPokemon) ? 2 : 1),
```
### Screenshots/Videos


https://discord.com/channels/1125469663833370665/1176874654015684739/1247953045489324122

## How to test the changes?




I tested the move by letting an ally dies and then i used retaliate move once, notice the high damages, then i used a second time the move retaliate and the move did less damage since no one died the previous turn. 
I used the overrides file but i do not introduced automated tests.

## Checklist
